python/tvm/relay/backend/contrib/ethosu/util.py

```diff
@@ -313,3 +318,37 @@ def __init__(
             encoded_constants,
             base_addresses,
         )
+
+
+def npu_pass(opt_level: int, name: str = ""):
```

Comment:
I would suggest to use an explicit name such as "create_npu_function_pass", because it has semantics equivalent to a function pass but only works with npu annotated functions.

Reply:
Sounds good, although I'm a bit hesitant to call it a function pass as the pass accepts an IRModule and returns an IRModule. Perhaps we should stick with `create_npu_pass`?

Reply:
I think that is the definition of the function pass.
See here : https://github.com/apache/tvm/blob/d4739eb80662a71781c46095dc2f3e1903b94f8c/src/relay/ir/transform.cc#L104-L105

So all passes are IRModule --> IRModule transformations (at least we should try to make them be :) )

Function passes one that mutates at function level but return the whole IRModule.

Reply:
In that case I'm okay with it :)